custom_components/goodwe/switch.py

```diff
@@ -18,72 +18,120 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-LOAD_CONTROL = SwitchEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class GoodweSettingsSwitchEntityDescription(SwitchEntityDescription):
+    """Required values when describing Goodwe switch settings."""
+
+    icon_on: str
```

Comment:
This should by done by icons translations see: https://developers.home-assistant.io/blog/2024/01/19/icon-translations

Reply:
it seems this is also a way: https://developers.home-assistant.io/docs/core/entity/#icon-property
I find it weird to extract this to a json and make things more complicated.

Reply:
Json is the preferred way in HA now.

Reply:
I can either remove this and keep a single icon (which will be in the code, like all other entities currently in this repo) or keep it like this; or if you are willing to push the change to the PR, that'd be nice

Reply:
Yes, I can prepare another commit for you.

Reply:
Hello, what is the status of this nice development ? thx